### ISSUE TITLE
poetry can't find git on Windows

### ISSUE THREAD

**Issue by neighthan** [sdispater/poetry#911]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**: 0.12.11 
- **pyproject.toml**:

```toml
[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"

[tool.poetry]
name = "wsl-wrappers"
description = "Wrappers for calling WSL commands from Windows."
version = "0.1.0"
authors = ["Nathan Hunt <neighthan.hunt@gmail.com>"]
license = "MIT"
readme = "README.md"
repository = "https://github.com/neighthan/wsl-wrappers"
include = ["pyproject.toml"]

[tool.poetry.scripts]
pdf2svg = "wsl_wrappers:scripts_api.pdf2svg"
pdflatex = "wsl_wrappers:scripts_api.pdflatex"

[tool.poetry.dependencies]
python = "^3.7"
toml = "^0.10.0"

[tool.poetry.dev-dependencies]
pytest = "^3.0"
twine = "^1.13"
invoke = {git = "https://github.com/neighthan/invoke.git"}
pytest-datafiles = "^2.0"
pre-commit = "^1.14"
pytest-cov = "^2.6"
```

## Issue

I just added a new dependency to my pyproject.toml which should be installed from GitHub (see `invoke` in the list above). Now my installing is failing:

```
>poetry install -vvv
Using virtualenv: C:\Users\neigh\AppData\Local\pypoetry\Cache\virtualenvs\wsl-wrappers-py3.7
Installing dependencies from lock file

[FileNotFoundError]
[WinError 2] The system cannot find the file specified

Exception trace:
 C:\Users\neigh\.poetry\lib\poetry\_vendor\py3.7\cleo\application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 C:\Users\neigh\.poetry\lib\poetry\console\application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 C:\Users\neigh\.poetry\lib\poetry\_vendor\py3.7\cleo\application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 C:\Users\neigh\.poetry\lib\poetry\console\commands\command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 C:\Users\neigh\.poetry\lib\poetry\_vendor\py3.7\cleo\commands\base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 C:\Users\neigh\.poetry\lib\poetry\_vendor\py3.7\cleo\commands\command.py in execute() at line 107
   return self.handle()
 C:\Users\neigh\.poetry\lib\poetry\console\commands\install.py in handle() at line 57
   return_code = installer.run()
 C:\Users\neigh\.poetry\lib\poetry\installation\installer.py in run() at line 76
   self._do_install(local_repo)
 C:\Users\neigh\.poetry\lib\poetry\installation\installer.py in _do_install() at line 216
   ops = solver.solve(use_latest=whitelist)
 C:\Users\neigh\.poetry\lib\poetry\puzzle\solver.py in solve() at line 38
   packages, depths = self._solve(use_latest=use_latest)
 C:\Users\neigh\.poetry\lib\poetry\puzzle\solver.py in _solve() at line 171
   self._package, self._provider, locked=locked, use_latest=use_latest
 C:\Users\neigh\.poetry\lib\poetry\mixology\__init__.py in resolve_version() at line 7
   return solver.solve()
 C:\Users\neigh\.poetry\lib\poetry\mixology\version_solver.py in solve() at line 79
   next = self._choose_package_version()
 C:\Users\neigh\.poetry\lib\poetry\mixology\version_solver.py in _choose_package_version() at line 354
   packages = self._provider.search_for(dependency)
 C:\Users\neigh\.poetry\lib\poetry\puzzle\provider.py in search_for() at line 132
   packages = self.search_for_vcs(dependency)
 C:\Users\neigh\.poetry\lib\poetry\puzzle\provider.py in search_for_vcs() at line 172
   git = Git()
 C:\Users\neigh\.poetry\lib\poetry\vcs\git.py in __init__() at line 37
   self._config = GitConfig(requires_git_presence=True)
 C:\Users\neigh\.poetry\lib\poetry\vcs\git.py in __init__() at line 16
   ["git", "config", "-l"], stderr=subprocess.STDOUT
 C:\Users\neigh\Anaconda3\lib\subprocess.py in check_output() at line 395
   **kwargs).stdout
 C:\Users\neigh\Anaconda3\lib\subprocess.py in run() at line 472
   with Popen(*popenargs, **kwargs) as process:
 C:\Users\neigh\Anaconda3\lib\subprocess.py in __init__() at line 775
   restore_signals, start_new_session)
 C:\Users\neigh\Anaconda3\lib\subprocess.py in _execute_child() at line 1178
   startupinfo)

install [--no-dev] [--dry-run] [-E|--extras EXTRAS] [--develop DEVELOP]
```

If I comment out the `invoke` line, installation succeeds, but I do get `RuntimeWarning: git executable could not be found` which must be the source of the install error.

However, just after that I can run on the command line:

```
>git --version
git version 2.19.1
```

Where is poetry looking for `git`? My `git` "installation" is, admittedly, somewhat odd, but since I can use it from the command line, why can't `poetry` find it?

**Comment by neighthan** [comment]
Hm, it seems like this error might come from `pip`, though poetry could have given some more information about it. Here's what happens with `pip`:

```
>pip install git+https://github.com/neighthan/invoke
Collecting git+https://github.com/neighthan/invoke
  Cloning https://github.com/neighthan/invoke to c:\users\neigh\appdata\local\temp\pip-req-build-jqe0wi2b
  Error [WinError 2] The system cannot find the file specified while executing command git clone -q https://github.com/neighthan/invoke C:\Users\neigh\AppData\Local\Temp\pip-req-build-jqe0wi2b
Cannot find command 'git' - do you have 'git' installed and in your PATH?
```

I'll leave this open because a better error message would at least be nice (e.g. passing through the whole message from `pip` not just `Error [WinError 2]`), but I think the real issue is upstream of `poetry`.

**Comment by Pluckerpluck** [comment]
> My `git` "installation" is, admittedly, somewhat odd

You're going to have to explain that. As a random example of why this could happen, you could have `git` actually be an alias or a function that's generated on shell startup, in which case `subprocess` can't actually find it because `git.exe` is not on your PATH.

Can you run `where.exe git`?

**Comment by neighthan** [comment]
My `git` is actually `git.cmd` as a wrapper for the `git` I have in Windows Subsystem for Linux:

```
>where git
C:\Users\neigh\Dropbox (MIT)\github\wsl-wrappers\wsl_wrappers\_scripts\git.cmd
```

I raised this issue in the `pip` repository (pypa/pip#6289), and it seems that it's because batch files (`.bat` or `.cmd`) are not treated the same way as `.exe` files in Windows and, in particular, can't be used by, e.g., `subprocess.run` unless you set `shell = True`. I fixed the problem by creating a simple `.exe` file which just calls my `.cmd` file (using [`shimmy`]). I'll close this now, since a workaround is present here; I still think the error on `poetry`'s side could be more descriptive, but this issue should come up if searching for it now.

[`shimmy`]: https://github.com/pfmoore/shimmy
